.github/actions/setup-node/action.yml

```
@@ -0,0 +1,36 @@
+name: 'Setup node.js'
+description: 'Setup node.js and cache dependencies'
+inputs:
+  node-version:
+    description: 'Node version'
+    required: true
+  os:
+    description: 'Operating system'
+    required: true
+runs:
+  using: 'composite'
+  steps:
+    - name: Cache node_modules
+      id: node-modules-cache
+      uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # v3.3.1
+      with:
+        path: node_modules
+        key: node_modules-${{ inputs.os }}-${{ inputs.node-version }}-${{ hashFiles('yarn.lock') }}
+
+    - name: Setup Node
+      uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+      with:
+        node-version: ${{ inputs.node-version }}
+        cache: ${{ steps.node-modules-cache.outputs.cache-hit == 'false' && 'yarn' || '' }}
+
+    - name: Install dependencies
+      uses: nick-fields/retry@943e742917ac94714d2f408a0e8320f2d1fcafcd # v2.8.3
+      if: steps.node-modules-cache.outputs.cache-hit == 'false'
+      with:
+        timeout_minutes: 10
+        max_attempts: 3
+        command: yarn install --frozen-lockfile --ignore-scripts
+
+    - name: Run scripts
+      shell: ${{ (inputs.os == 'Windows' && 'pwsh') || 'bash' }}
```

```suggestion
      shell: bash
```
why changing the shell? bash should work on windows, but a single command should also work on cmd

---

Composite actions are very limited, in particular they require shell to be set explicitly. Also, there is no step outputs access when called from the same repository, so I'm about to extract it to `renovatebot/setup-node`.

---

Seems like I should use reusable workflows instead

---

No, reusable workflows use another runner instance, while we want composition at steps level